python/hsfs/core/feature_view_engine.py

```diff
@@ -61,6 +62,11 @@ def __init__(self, feature_store_id):
         self._query_constructor_api = query_constructor_api.QueryConstructorApi()
 
     def save(self, feature_view_obj):
+        if feature_view_obj.query.is_time_travel():
```

Comment:
I think the check should be recursive for all sub-queries within the query. If, e.g., the right side of the join is a time-travel query we should catch that as well.

Reply:
Since we set `is_time_travel` from the top level, so it should be sufficient to check only the top level, right?